go/libraries/doltcore/env/actions/branch.go

```diff
@@ -187,105 +205,144 @@ func CreateBranchOnDB(ctx context.Context, ddb *doltdb.DoltDB, newBranch, starti
 	return ddb.NewBranchAtCommit(ctx, newRef, cm)
 }
 
-func CreateBranch(ctx context.Context, dEnv *env.DoltEnv, newBranch, startingPoint string, force bool) error {
-	return CreateBranchOnDB(ctx, dEnv.DoltDB, newBranch, startingPoint, force, dEnv.RepoState.CWBHeadRef())
+func createBranch(ctx context.Context, dbData env.DbData, newBranch, startingPoint string, force bool) error {
+	return CreateBranchOnDB(ctx, dbData.Ddb, newBranch, startingPoint, force, dbData.Rsr.CWBHeadRef())
 }
 
-func CheckoutBranch(ctx context.Context, dEnv *env.DoltEnv, brName string) error {
-	dref := ref.NewBranchRef(brName)
-
-	hasRef, err := dEnv.DoltDB.HasRef(ctx, dref)
+// getCheckoutHashes writes the roots needed for a checkout and returns the updated work and staged hash.
+func getCheckoutHashes(ctx context.Context, dbData env.DbData, dref ref.DoltRef, brName string) (wrkHash hash.Hash, stgHash hash.Hash, err error) {
```

Comment:
Seems like this should still be called CheckoutBranch or similar?

As a general rule, if a method is called getSomething and it writes to disk, that's a bad name

Reply:
Maybe updateRootsForBranch?

Reply:
That's much better.